### DISCUSSION TITLE
[Components] Progress tracker

### DISCUSSION THREAD

**Discussion by soykje** [adevinta/spark#1671] [Ideas]
**Specs:** [Figma](https://www.figma.com/file/0QchRdipAVuvVoDfTjLrgQ/Spark-Specs-and-Handoff?type=design&node-id=12220-13089&mode=design&t=PmEST0Vr3GmVhyIs-0)

**References:**
- [Atlassian](https://atlassian.design/components/progress-tracker/examples)
- [Aurora](https://design.gccollab.ca/component/progress-indicators/)
- [Carbon DS](https://carbondesignsystem.com/components/progress-indicator/usage/)
- [Dell DS](https://www.delldesignsystem.com/components/progress-tracker/#anatomy)
- [W3C DS](https://design-system.w3.org/styles/forms.html#progress-indicators)

**Expected features and notes:**
- user should be able to switch interactive mode
- for a11y reasons we should provide labels (`aria-label` at least)

#### `ProgressTracker`
- **`onStepClick`**: if defined, the `ProgressTracker` will be interactive. Component wrapper will then switch from simple and generic `div` element to more specific `nav` one.
- **`stepIndex`**: the index of the current step.
- **`orientation`**: could be `vertical` or `horizontal`.

#### `ProgressTracker.Step`
_Icon and counter should be internally handled thanks to the component index in the declared ordered list._
- **`label`**: each `ProgressTracker.Step` should be provided a label (`string` type) or at least an `aria-label`.

```tsx
<ProgressTracker>
  <ProgressTracker.Step label="Step one" />
  <ProgressTracker.Step label="Step two" />
  <ProgressTracker.Step label="Step three" />
</ProgressTracker>
```
Based on specifications, we also may want to implement other props such as `disabled`, `intent`, ...

**Comment by acd02** [comment]
Regarding the **labels**, how should we handle them when the text is really long?
Also, what about their placements in relation to the progress indicators? Should we allow the possibility of placing them at the bottom or top of it, or should we only allow them to be placed at the bottom? (I think this is a question that should be addressed by the design team)

**Reply by soykje** [reply]
At this moment the discussion with Design team hasn't been done, but we should meet soon to discuss and specify that :+1: Anyway, I don't think it would be too structuring here, wdyt?
